src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -51,46 +51,51 @@
  * @author Peter Güttinger
  */
 @Name("Tool")
-@Description("The item a player is holding.")
-@Examples({"player is holding a pickaxe",
-		"# is the same as",
-		"player's tool is a pickaxe",
-		"player's off hand tool is shield #Only for Minecraft 1.9"})
+@Description("The item an entity is holding in their main or off hand.")
+@Examples({"player's tool is a pickaxe",
+	"player's off hand tool is a shield",
+	"set tool of all player's to diamond sword",
+	"set offhand tool of target entity to a bow"})
 @Since("1.0")
 public class ExprTool extends PropertyExpression<LivingEntity, Slot> {
 	static {
-		Skript.registerExpression(ExprTool.class, Slot.class, ExpressionType.PROPERTY, "[the] (tool|held item|weapon) [of %livingentities%]", "%livingentities%'[s] (tool|held item|weapon)");
+		Skript.registerExpression(ExprTool.class, Slot.class, ExpressionType.PROPERTY,
+			"[the] (0¦(tool|held item|weapon)|1¦(off[ ]hand (tool|item)|shield[ item])) [of %livingentities%]",
+			"%livingentities%'[s] (0¦(tool|held item|weapon)|1¦(off[ ]hand (tool|item)|shield[ item]))");
 	}
-	
+
+	private boolean offHand;
+
 	@SuppressWarnings({"unchecked", "null"})
 	@Override
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parser) {
 		setExpr((Expression<Player>) exprs[0]);
+		offHand = parser.mark == 1;
 		return true;
 	}
-	
+
 	@Override
 	protected Slot[] get(final Event e, final LivingEntity[] source) {
 		final boolean delayed = Delay.isDelayed(e);
 		return get(source, new Getter<Slot, LivingEntity>() {
 			@Override
 			@Nullable
-			public Slot get(final LivingEntity p) {
+			public Slot get(final LivingEntity ent) {
 				if (!delayed) {
-					if (e instanceof PlayerItemHeldEvent && ((PlayerItemHeldEvent) e).getPlayer() == p) {
+					if (e instanceof PlayerItemHeldEvent && ((PlayerItemHeldEvent) e).getPlayer() == ent) {
 						final PlayerInventory i = ((PlayerItemHeldEvent) e).getPlayer().getInventory();
 						assert i != null;
 						return new InventorySlot(i, getTime() >= 0 ? ((PlayerItemHeldEvent) e).getNewSlot() : ((PlayerItemHeldEvent) e).getPreviousSlot());
-					} else if (e instanceof PlayerBucketEvent && ((PlayerBucketEvent) e).getPlayer() == p) {
+					} else if (e instanceof PlayerBucketEvent && ((PlayerBucketEvent) e).getPlayer() == ent) {
 						final PlayerInventory i = ((PlayerBucketEvent) e).getPlayer().getInventory();
 						assert i != null;
```

Comment:
these asserts can be removed as HumanEntity#getInventory is marked with NotNull now

Reply:
cool. Removed